app/src/main/java/com/cornellappdev/coffee_chats_android/SchedulingActivity.kt

```diff
@@ -0,0 +1,101 @@
+package com.cornellappdev.coffee_chats_android
+
+import android.os.Bundle
+import android.view.View
+import android.widget.*
+import androidx.appcompat.app.AppCompatActivity
+import androidx.constraintlayout.widget.ConstraintLayout
+import kotlinx.android.synthetic.main.activity_scheduling.*
+import android.widget.AdapterView.OnItemClickListener
+import kotlinx.android.synthetic.main.activity_scheduling.back_button
+import kotlinx.android.synthetic.main.fragment_create_profile.*
+
+
+class SchedulingActivity : AppCompatActivity() {
+    var currDay: String = "Sunday"
+    lateinit var currDayTextView: TextView
+    private val days = arrayOf("Su", "M", "Tu", "W", "Th", "F", "Sa")
+    private val daysFullName = arrayOf("Sunday", "Monday", "Tuesday", "Wednesday", "Thursday",
+        "Friday", "Saturday")
+    private val times = arrayOf("9:00", "1:00", "5:00", "9:30", "1:30", "5:30", "10:00", "2:00",
+    "6:00", "10:30", "2:30", "6:30", "11:00", "3:00", "7:00", "11:30", "3:30", "7:30", "12:00",
+    "4:00", "8:00", "12:30", "4:30", "8:30")
+    var selectedTimes  = HashMap<String, MutableList<String>>()
+    var selectedDays = mutableSetOf<String>()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_scheduling)
+
+        val timeAdapter = TimeOptionAdapter(this, times)
+        time_gridview.adapter = timeAdapter
+        var dayAdapter = DayAdapter(this, days)
+        day_selection.adapter = dayAdapter
+
+        day_header.text = "Every Sunday"        // Sunday by default
+        for (day in daysFullName) {
+            selectedTimes[day] = mutableListOf<String>()
+        }
+
+        back_button.setOnClickListener { finish() }
+
+        var previousDot: ImageView? = null
+        day_selection.onItemClickListener = OnItemClickListener { parent, v, position, id ->
+//            val daySelectedIndex = dayAdapter.selectedPositions.indexOf(position)
+            val daySelectedView = day_selection.getChildAt(position) as ConstraintLayout
+            val daySelectedTextView = daySelectedView.getChildAt(0) as TextView
+            val daySelectedDot = daySelectedView.getChildAt(1) as ImageView
+            val sundaySelectedView = day_selection.getChildAt(0) as ConstraintLayout
+            val sundaySelectedDot = sundaySelectedView.getChildAt(1) as ImageView
```

Comment:
please add a comment for this chunk of code, or at least space out the logic with a newline